pinot-controller/src/main/java/org/apache/pinot/controller/api/upload/ZKOperator.java

```diff
@@ -204,8 +205,8 @@ private void checkCRC(HttpHeaders headers, String offlineTableName, String segme
   }
 
   private void processNewSegment(SegmentMetadata segmentMetadata, URI finalSegmentLocationURI,
-      File currentSegmentLocation, String zkDownloadURI, String crypter, String tableNameWithType, String segmentName,
-      boolean moveSegmentToFinalLocation) {
+      File currentSegmentLocation, String zkDownloadURI, HttpHeaders headers, String crypter, String tableNameWithType,
```

Comment:
I think `headers` is nullable?

Reply:
Yes, I think you are right. Updated to handle `null` case.